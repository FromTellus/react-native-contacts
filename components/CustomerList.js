import {View, FlatList, Button, StyleSheet, Text} from 'react-native';
import {useState} from 'react';
import {customers} from '../customers';

export default CustomerList = ({navigation}) => {
  const [user, setUser] = useState(customers);

  return (
    <View>
      <FlatList
        keyExtractor={(item, index) => index.toString()}
        contentContainerStyle={styles.contactList}
        data={user}
        renderItem={({item}) => (
          <Text
            onPress={() => navigation.navigate('Details', item)}
            style={styles.contactItem}
          >
            {item.name.first} Phone : {item.phone}
          </Text>
        )}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  contactList: {
    justifyContent: 'center',
    alignItems: 'left',
    marginTop: 0,
    paddingVertical: 10,
    backgroundColor: 'white',
    color: '#20232a',
    textAlign: 'center',
    fontSize: 30,
    fontWeight: 'bold',
    minHeight: '100%',
  },
  contactItem: {
    fontSize: 20,
    padding: 5,
    margin: 5,
    borderWidth: 2,
    borderColor: 'black',
    borderRadius: 6,
    minWidth: '100%',
    backgroundColor: '#DBE8D8',
  },
});
