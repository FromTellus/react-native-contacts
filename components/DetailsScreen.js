import {View, Image, Text, StyleSheet, TextInput, Button} from 'react-native';
import { useState } from 'react';

export default function DetailsScreen({ route, navigation }) {
    const { name, location, email, phone, picture } = route.params;
    const [comment, setComment] = useState("");
  
    return (
      <>
        <View style={styles.contact}>
          <View style={styles.contactInformation}>
            <Image
              source={{
                uri: picture.large,
                width: 200,
                height: 200,
              }}
            />
            <Text>
              Personalia : {name.title} {name.first} {name.last}
            </Text>
            <Text>Phone number: {phone}</Text>
            <Text>Email address : {email}</Text>
            <Text>
              Located in : {location.country}, {location.city}
            </Text>
            <Text>Comments : {comment} </Text>
          </View>
  
          <View>
            <Text>Write your comment</Text>
            <TextInput
              style={{ backgroundColor: "lightgray" }}
              placeholder="Type your comment here ..."
              multiline={true}
              numberOfLines={4}
              onSubmitEditing={2}
            />
          </View>
          <Button title="Submit comment"></Button>
        </View>
      </>
    );
  }

  const styles = StyleSheet.create({
    contact: {
      alignItems: "center",
      padding: 10,
      borderColor: "black",
      borderRadius: 6,
      backgroundColor: "#DBE8D8",
      fontSize: 50,
      color: "white",
      minHeight: "100%",
    },
    contactInformation: {
        alignItems: "center",
        padding: 10,
        borderColor: "black",
        borderRadius: 6,
        fontSize: 50,
        color: "white",
        minHeight: "100%",
      },
  });