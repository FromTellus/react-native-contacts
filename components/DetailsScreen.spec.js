import renderer from "react-test-renderer";
import DetailsScreen from "./DetailsScreen";
import CustomerList from "./CustomerList";

// check that detail screen is getting image and data from CustomerList

describe("<DetailsScreen />", () => {
  const route = {
    route: {
      params: {
        name: {
          first: "John",
          last: "Doe",
          title: "Mr",
        },
        address: "123 Main St",
        city: "Anytown",
        picture: {
          large: "https://randomuser.me/api/portraits/men/34.jpg",
          medium: "https://randomuser.me/api/portraits/med/men/34.jpg",
          thumbnail: "https://randomuser.me/api/portraits/thumb/men/34.jpg",
        },
        location: {
          country: "sweden",
          city: "Stockholm"
        },
      },
    },
  };

  it("renders correctly", () => {
    const tree = renderer.create(<DetailsScreen {...route} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
