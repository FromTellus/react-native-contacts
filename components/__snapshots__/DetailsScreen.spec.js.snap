// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#DBE8D8",
      "borderColor": "black",
      "borderRadius": 6,
      "color": "white",
      "fontSize": 50,
      "minHeight": "100%",
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "black",
        "borderRadius": 6,
        "color": "white",
        "fontSize": 50,
        "minHeight": "100%",
        "padding": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "height": 200,
          "uri": "https://randomuser.me/api/portraits/men/34.jpg",
          "width": 200,
        }
      }
    />
    <Text>
      Personalia : 
      Mr
       
      John
       
      Doe
    </Text>
    <Text>
      Phone number: 
    </Text>
    <Text>
      Email address : 
    </Text>
    <Text>
      Located in : 
      sweden
      , 
      Stockholm
    </Text>
    <Text>
      Comments : 
       
    </Text>
  </View>
  <View>
    <Text>
      Write your comment
    </Text>
    <TextInput
      multiline={true}
      numberOfLines={4}
      onSubmitEditing={2}
      placeholder="Type your comment here ..."
      style={
        Object {
          "backgroundColor": "lightgray",
        }
      }
    />
  </View>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Submit comment
      </Text>
    </View>
  </View>
</View>
`;
